src/Kiota.Builder/Caching/DocumentCachingProvider.cs

```diff
@@ -37,10 +37,10 @@ public class DocumentCachingProvider {
                 Directory.CreateDirectory(directory);
             Stream content = null;
             try {
-                using var httpContent = await HttpClient.GetStreamAsync(documentUri, token);
+                await using var httpContent = await HttpClient.GetStreamAsync(documentUri, token);
```

Comment:
Can you provide some additional context on why we're awaiting twice here? Is this a new syntax to use dispose async instead?

Reply:
Yes, exactly. Types can implement the `IAsyncDisposable` interface additionally to `IDisposable`. This allows disposing of types in a non-blocking manner. `await using` is the syntax which tells the runtime to call `DisposeAsync()` instead of `Dispose()`.

Reply:
Thanks for confirming, I wasn't aware of this new syntax (only of the disposeasync method). We most likely have other occurrences in the code

Reply:
I can take a look later and open another PR.